sdk/python/feast/infra/aws_dynamo_provider.py

```diff
@@ -0,0 +1,201 @@
+from datetime import datetime
+from typing import Any, Callable, Dict, List, Optional, Sequence, Tuple, Union
+
+import boto3
+import mmh3
+import pandas
+from botocore.exceptions import ClientError
+
+from feast import FeatureTable, utils
+from feast.feature_view import FeatureView
+from feast.infra.key_encoding_utils import serialize_entity_key
+from feast.infra.offline_stores.helpers import get_offline_store_from_sources
+from feast.infra.provider import (
+    Provider,
+    RetrievalJob,
+    _convert_arrow_to_proto,
+    _get_column_names,
+    _run_field_mapping,
+)
+from feast.protos.feast.types.EntityKey_pb2 import EntityKey as EntityKeyProto
+from feast.protos.feast.types.Value_pb2 import Value as ValueProto
+from feast.registry import Registry
+from feast.repo_config import DynamoOnlineStoreConfig, RepoConfig
+
+
+class AwsDynamoProvider(Provider):
```

Comment:
shall we name this AwsDynamodbProvider instead? [Dynamo](https://www.allthingsdistributed.com/2007/10/amazons_dynamo.html) is not exactly the same as Dynamo DB